src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonObject.cs

```diff
@@ -78,7 +78,7 @@ internal JsonObject(JsonElement element, JsonNodeOptions? options = null) : this
         /// <returns>
         ///   <see langword="true"/> if a property with the specified name was found; otherwise, <see langword="false"/>.
         /// </returns>
-        public bool TryGetPropertyValue(string propertyName, out JsonNode? jsonNode) =>
+        public bool TryGetPropertyValue(string propertyName, [NotNullWhen(true)] out JsonNode? jsonNode) =>
```

Comment:
won't this return null if property exists but it's literally assigned to `null`? can you add a simple test that is (or not) a case?